### PR TITLE
Next - Replay unpacker rewrite

### PR DESCRIPTION
**Pull request by stewieoO**

### CODE REVIEW

**Review by SakuraIsayeki — APPROVED** [review]
An excellent rewrite, that addresses many of the current issues and expandability points that the library was facing. 

As it is, documentation is clearly lacking, and we cannot release a production Nupkg in this state. This is something we'll have to address before any mainstream release.
However, I'm all in favour of releasing this as a preview version, post-merge, pending further documentation.

File: Nodsoft.WowsReplaysUnpack/Controllers/ReplayControllerBase.cs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+{
+	private static readonly Dictionary<string, MethodInfo[]> _methodSubscriptions;
+	private static readonly Dictionary<string, MethodInfo[]> _propertyChangedSubscriptions;
+	protected IDefinitionStore DefinitionStore { get; }
+	protected ILogger<Entity> EntityLogger { get; }
+
+	public UnpackedReplay Replay { get; protected set; }
+
+	static ReplayControllerBase()
+	{
```
Comment (floribe2000) [inline Nodsoft.WowsReplaysUnpack/Controllers/ReplayControllerBase.cs:23]: Consider to replace with source generation later if necessary due to performance

File: Nodsoft.WowsReplaysUnpack/Controllers/ReplayControllerBase.cs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+		ILogger<Entity> entityLogger)
+	{
+		DefinitionStore = definitionStore;
+		EntityLogger = entityLogger;
+	}
+	
+#pragma warning restore CS8618
+	
+	
+	public virtual UnpackedReplay CreateUnpackedReplay(ArenaInfo arenaInfo)
```
Comment (floribe2000) [inline Nodsoft.WowsReplaysUnpack/Controllers/ReplayControllerBase.cs:55]: Can this be called twice? If yes, maybe use the existing element?

File: Nodsoft.WowsReplaysUnpack/Controllers/ReplayControllerBase.cs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+	{
+		Replay = new(arenaInfo);
+		return Replay;
+	}
+
+	#region Packet Handling
+	public virtual void HandleNetworkPacket(NetworkPacketBase networkPacket, ReplayUnpackerOptions options)
+	{
```
Comment (floribe2000) [inline Nodsoft.WowsReplaysUnpack/Controllers/ReplayControllerBase.cs:63]: Shouldn't this be a switch?

File: Nodsoft.WowsReplaysUnpack.Benchmark/Program.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Environments;
+using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Running;
+using Nodsoft.WowsReplaysUnpack.Benchmark;
+
+BenchmarkRunner.Run<UnpackBenchmark>(DefaultConfig.Instance
+	.WithOptions(ConfigOptions.DisableOptimizationsValidator)
+	.AddJob(Job.Default.WithRuntime(CoreRuntime.Core50))
```
Comment (floribe2000) [inline Nodsoft.WowsReplaysUnpack.Benchmark/Program.cs:11]: Remove .NET 5 benchmark now that .NET 5 support has been dropped

**Review by floribe2000 — CHANGES_REQUESTED** [review]
Overalll, a very good change.
However, there are some minor things where I left comments, please review them before this can be merged.

Is there any way to get rid of the Microsoft.Extensions.Logging dependency for the core library? This library may be used in situations where there is no static host and therefore no simple MS dependency injection available, one of these cases is using the library in an Avalonia application.

Also, update the Readme before this can be merged into dev and add credits stewie to the list of authors.
Also consider using a Directory.Build.props file to synchronize the version and author list between subprojects like the library projects.

**Review by floribe2000 — APPROVED** [review]
LGTM

**Pull request merged**